js/classroom.js

```diff
@@ -0,0 +1,207 @@
+  // conflicts because it is active (obviously). Also there are only conflicts
+  // if there is a combination being displayed.
+  if (this != activeClassroom && lecture.parent.activeCombinationIndex != null) {
+    var activeCombination = lecture.parent.combinations[lecture.parent.activeCombinationIndex];
+    if (activeCombination.length == 0) {
+      // This happens when there are at least one lecture displayed on the screen
+      // but no lecture is selected (div#lecture-schedule is graphically empty).
+      // However, in this situation there may be various lectures included,
+      // i.e. plan.lectures.length >>> 0. 
+      // So there is no conflict, we can return.
+      return
+    }
+    var lecturesClassroom = activeCombination.lecturesClassroom;
+    for (var i = 0; i < lecturesClassroom.length; i++) {
+      if (this.parent == lecturesClassroom[i].parent) {
+        // Same lecture, skip.
+        continue;
+      }
+      // Iterate over every schedule on this classroom and every active classroom
+      // on the current active combination.
+      for (var j = 0; j < this.schedules.length; j++) {
+        for (var k = 0; k < lecturesClassroom[i].schedules.length; k++) {
+          if (schedulesConflict(this.schedules[j], lecturesClassroom[i].schedules[k])) {
+            // This schedule (one of many for this classroom) conflicts with some other
+            // schedule from an active classroom. Set conflict highlight.
+            addClass(this.schedules[j].htmlElement, 'schedule-box-highlight-conflict');
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ *
+ */
+Classroom.prototype.unsetHighlight = function() {
+  var lecture = this.parent;
+  if (lecture.activeClassroomIndex != null) {
+    // There is an active classroom for this lecture.
+    var activeClassroom = lecture.classrooms[lecture.activeClassroomIndex];
+    activeClassroom.showBox();
+  }
+  this.removeClassInSchedules('schedule-box-highlight');
+  this.removeClassInSchedules('schedule-box-highlight-conflict');
+}
+
+/**
+ *
+ */
+Classroom.prototype.toggleClassroomSelection = function() {
+  toggleClass(this.htmlElement, 'classroom-selected');
+  this.selected = !this.selected;
+
+  var thisLecture = this.parent;
+  var noClassroomSelected = true;
+  for (var i = 0; i < thisLecture.classrooms.length; i++) {
+    if (thisLecture.classrooms[i].selected) {
+      noClassroomSelected = false;
+    }
+  }
+
+  if (noClassroomSelected) {
+    thisLecture.selected = false;
+    thisLecture.activeClassroomIndex = null;
+  } else {
+    thisLecture.selected = true;
+  }
+
+  // Update this plan.
+  thisLecture.parent.update(this);
+}
+
+
+/**
+ * This function adds event listeners to 'mouseenter', 'mouseleave' and 'click'
+ *
+ * @see Classroom#setHighlight
+ * @see Classroom#unsetHighlight
+ * @see Classroom#toggleClassroomSelection
+ */
+Classroom.prototype.addEventListeners = function() {
```

Comment:
Performance: We probably will need `removeEventListeners()` and maybe a `dispose()` method to call. This is to avoid memory leaks.